packages/lit-analyzer/src/rules/util/type/is-assignable-in-event-binding.ts

```diff
@@ -0,0 +1,129 @@
+import { isAssignableToSimpleTypeKind, SimpleType, typeToString, validateType } from "ts-simple-type";
+import { parseSimpleJsDocTypeExpression } from "web-component-analyzer";
+import { HtmlNodeAttr } from "../../../analyze/types/html-node/html-node-attr-types";
+import { RuleModuleContext } from "../../../analyze/types/rule/rule-module-context";
+import { rangeFromHtmlNodeAttr } from "../../../analyze/util/range-util";
+import { isAssignableToType } from "./is-assignable-to-type";
+
+export function isAssignableInEventBinding(
+	htmlAttr: HtmlNodeAttr,
+	{ typeA, typeB }: { typeA: SimpleType; typeB: SimpleType },
+	context: RuleModuleContext
+): boolean | undefined {
+	// Make sure that the expression given to the event listener binding a function or an object with "handleEvent" property.
+	// This catches typos like: @click="onClick()"
+	if (!isTypeBindableToEventListener(typeB)) {
+		context.report({
+			location: rangeFromHtmlNodeAttr(htmlAttr),
+			message: `You are setting up an event listener with a non-callable type '${typeToString(typeB)}'`
+		});
+
+		return false;
+	}
+
+	return isEventAssignable(htmlAttr, { typeA, typeB }, context);
+}
+
+export function isEventAssignable(
+	htmlAttr: HtmlNodeAttr,
+	{ typeA, typeB }: { typeA: SimpleType; typeB: SimpleType },
+	context: RuleModuleContext
+): boolean | undefined {
+	// Treat type "ANY" as "Event"
+	if (typeA.kind === "ANY") {
+		typeA = parseSimpleJsDocTypeExpression("Event", context);
+	}
+
+	// Construct an event handler type to perform type checking against
+	const expectedType: SimpleType = {
+		kind: "OBJECT",
```

Comment:
would you mind explaining what this layer is representing? and why it isn't just `kind: FUNCTION` at top level?

Reply:
Typescript represents the type of a function as an object with a call signature. This could just as well be a `kind: FUNCTION` at top level as it would type check the same, but I wanted to put in the exact same type as what I get when analyzing an event handler.

Reply:
aha learn something new every day, thanks for the explanation :+1: